### PR TITLE
Dockerizing Argum Server

### PR DESCRIPTION
**Pull request by M-Wittner**
Release & Dev Dockerfiles

Updated README with Docker instructions.

### CODE REVIEW

File: docker-entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/sh
```
Comment (benny-n) [inline docker-entrypoint.sh:1]: This file seems pretty redundant

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+USER appuser
+# Copy the binary from the builder stage.
+COPY --from=builder /app/target/release/argum-server .
+
+COPY ./docker-entrypoint.sh .
+
+# Expose the port that the application listens on.
+EXPOSE ${PORT}
+
+ENTRYPOINT [ "./docker-entrypoint.sh" ]
```
Comment (benny-n) [inline Dockerfile:43]: I think just running ./argum-server is enough as an entry point, as mentioned above

File: README.md [hunk 3]
```diff
@@ -17,3 +17,34 @@ For a list of environment variables, see the [`.env.example`](.env.example) file
...
+```
+
+Run Dev Docker image
+```bash
+docker run --rm -it -p 3000:3000 -v $PWD:/app argum-server
+```
+
+Start the server
+```bash
+cargo ran
```
Comment (benny-n) [inline README.md]: `cargo run`

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    apk upgrade
+
+# Install the musl target so we can build static binaries.
+RUN apk add --no-cache musl-dev
+
+COPY Cargo.* ./
+COPY src ./src
+
+# Build the application.
+RUN cargo build --release
```
Comment (benny-n) [inline Dockerfile:16]: This is probably ok, but just FYI the `--release` flag creates an optimized build and therefore leads to much longer compile times.
Ideally, for development we would just want to run `cargo build` without the release flag, but it doesn't matter too much.

**Review by benny-n — CHANGES_REQUESTED** [review]

File: docker-entrypoint.sh (hunk 1 above)
Comment (M-Wittner) [reply docker-entrypoint.sh:1]: That's a common good practice to have an entry point script rather than running the binary.

File: README.md (hunk 3 above)
Comment (M-Wittner) [reply README.md]: Oopsi 😊

File: Dockerfile (hunk 4 above)
Comment (M-Wittner) [reply Dockerfile:16]: Please note that there are 2 Dokerfiles as mentioned above, one for development which doesn't care for the release flag

File: Dockerfile (hunk 2 above)
Comment (M-Wittner) [reply Dockerfile:43]: Same as the above comment. +
The server had no indication it start and actuality runs so the entrypoint for the least, gives an indication what's up.

Comment (benny-n) [reply Dockerfile:43]: Okay, I see your point, but IMO the solution should be to enable logging from the server code, and then get the indication from there, not from another file.
Anyway it does not matter to me as much so we can leave it until we enable proper logging from the server

File: Dockerfile (hunk 4 above)
Comment (benny-n) [reply Dockerfile:16]: Are you sure?
I only see 4 files in the changes:
 - .dockerignore
 - Dockerfile
 - README.md
 - docker-entrypoint.sh

Comment (M-Wittner) [reply Dockerfile:16]: Damn, I forgot to add it -_-
My bad. pushed.

**Review by benny-n — APPROVED** [review]

**Pull request merged**